File: palindrome_simple/palindrome_num.cpp
#include <stdio.h>
#include <iostream>
#include <math.h>
#include <string>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        string num = to_string(x);
        int len = num.size();
        if (x<0){
            return false;
        }
        int x_cpy = x;
        int i = 0, new_num = 0;
        while (i < len) {
            new_num = new_num*10 + x_cpy%10;
            x_cpy = int(x_cpy / 10);
            i += 1;
        }
        if (x == new_num){
            return true;
        }
        return false;
    }
};

File: palindrome_simple/palindrome_num.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        new_num = 0
        len_num = len(str(x))
        cpy_x = x
        i = 0
        while i < len_num:
            new_num = new_num*10 + cpy_x%10 
            cpy_x //= 10
            i += 1
        
        if new_num == x:
            return True
        return False